From b4da5b852329d5f67d6a48ff44e188e366adc3d2 Mon Sep 17 00:00:00 2001
From: Paul Naoumenko <paulx.naoumenko@intel.com>
Date: Fri, 18 Oct 2013 20:29:13 +0200
Subject: [PORT FROM MAIN] wifi: restore original SoftAP controller behavior

BZ: 107357

During bring-up phase, some hacks were made to turn on and off
wifi interfaces during wifi to hotspot switch.
This is not needed anymore and was slowing down wifi ON/OFF
cycles.
Also remove 11n mode parameter as it's not taken into account anyway
if hw_mode is not specified. Also 4335 is 11ac so specifying 11n may
give less throughput

Change-Id: I7da51644a7bde2b09402d4621c3a1c88cfb8516c
Orig-Change-Id: I7dd3d61081fe3addc035a503948b57b647ea4b7d
Category: device enablement
Domain: CWS.WIFI-Common
Origin: upstream
Upstream-Candidate: no,proprietary
Signed-off-by: Jonathan DE CESCO <jonathan.de.cesco@intel.com>
---
 SoftapController.cpp |   12 +-----------
 1 files changed, 1 insertions(+), 11 deletions(-)

diff --git a/SoftapController.cpp b/SoftapController.cpp
index 5fc4f72..e3cbc75 100644
--- a/SoftapController.cpp
+++ b/SoftapController.cpp
@@ -180,7 +180,7 @@ int SoftapController::setSoftap(int argc, char *argv[]) {
 
 
     asprintf(&wbuf, "interface=%s\ndriver=nl80211\nctrl_interface="
-            "/data/misc/wifi/hostapd\nssid=%s\nchannel=%d\nieee80211n=1\n"
+            "/data/misc/wifi/hostapd\nssid=%s\nchannel=%d\n"
             "hw_mode=g\nignore_broadcast_ssid=%d\n",
             argv[2], argv[3], channel, hidden);
 
@@ -257,9 +257,6 @@ int SoftapController::fwReloadSoftap(int argc, char *argv[])
         return ResponseCode::CommandSyntaxError;
     }
 
-    iface = argv[2];
-    stopDriver(iface);
-
     if (!strcmp("STA", argv[3]))
         wifi_switch_driver_mode(WIFI_STA_MODE); /* which is STA + P2P... */
     else if (!strcmp("AP", argv[3]))
@@ -267,13 +264,6 @@ int SoftapController::fwReloadSoftap(int argc, char *argv[])
     else if (!strcmp("P2P", argv[3]))
         wifi_switch_driver_mode(WIFI_P2P_MODE);
 
-    /**
-     * Sleep to workaround issue in the brcm driver which is tracked by BZ
-     * 85864. To be removed as soon as BZ85864 is fixed.
-     */
-    sleep(2);
-    startDriver(iface);
-
     ALOGD("Softap fwReload - done");
 
     return ResponseCode::SoftapStatusResult;
-- 
1.7.4.1

